### PR TITLE
Pract.03>: Add 1 - 17 | Georgi Vasilev,  Pract.04: Add 1 - 22 | Georgi Vasilev

### PR DESCRIPTION
**Pull request by GeorgiVasilevv**
Georgi Vasilev
3 0MI0600499

### CODE REVIEW

File: Sem.04/Pract.04/Georgi Vasilev/E12.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// E12.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
+
+#include <iostream>
+//12
+int leastCommonMultiple(int a, int b, int c) {
+
+	// �������� ���1 �� ��� �����
```
Comment (SirLex) [inline Sem.04/Pract.04/Georgi Vasilev/E12.cpp:8]: interesting

File: Sem.04/Pract.04/Georgi Vasilev/E18.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+//18
+int getNumberInRange(int from, int to) {
+	int i; std::cin >> i;
+
+	while (i < from || i > to)
+	{
+		std::cin >> i;
+	}
+	return i;
+
```
Comment (SirLex) [inline Sem.04/Pract.04/Georgi Vasilev/E18.cpp:15]: non needed new lines everywhere. Use them to make the code clearer to read

File: Sem.04/Pract.04/Georgi Vasilev/E20.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+}
+long long CalculateDigitSums(long long n) {
+	long long sum = digitSum(n);
+	while (sum >= 10)
+	{
+		sum = digitSum(sum);
+	}
+	return sum;
+
+}
```
Comment (SirLex) [inline Sem.04/Pract.04/Georgi Vasilev/E20.cpp:23]: and a lot of places where you miss new lines after functions

File: Sem.04/Pract.04/Georgi Vasilev/E12.cpp (hunk 1 above)
Comment (GeorgiVasilevv) [reply Sem.04/Pract.04/Georgi Vasilev/E12.cpp:8]: int leastCommonMultiple(int a, int b, int c) {

	
	int a1 = a;
	int b1 = b;

	if (b > a)
	{
		int temp = a;
		a = b;
		b = temp;
	}
	while (b != 0)
	{
		int oldB = b;
		b = a % b;
		a = oldB;
	}

	
	int NOK1 = (a1 * b1) / a;
	a = NOK1;

	
	int a2 = a;
	int c1 = c;

	if (c > a)
	{
		int temp = a;
		a = c;
		c = temp;
	}
	while (c != 0) {
		int oldC = c;
		c = a % c;
		a = oldC;
	}

	
	int NOK2 = (a2 * c1) / a;
	return NOK2;
}

**Pull request merged**